src/Controller/JobController.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Controller;
+
+use App\Entity\Company;
+use App\Entity\Jobs;
+use App\Repository\CompanyRepository;
+use App\Repository\JobsRepository;
+use App\Validator\JobValidator;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use DateTimeImmutable;
+
+/**
+ * new class for company controller
+ */
+class JobController extends AbstractController
+{
+
+    private JobsRepository $jobsRepository;
+    private JobValidator $jobValidator;
+    private CompanyRepository $companyRepository;
+   
+    /**
+     * __construct
+     *
+     * @param  mixed $companyRepository
+     * @param  mixed $companyValidator
+     * @return void
+     */
+    public function __construct(JobsRepository $jobsRepository, JobValidator $jobValidator, CompanyRepository $companyRepository)
+    {
+        $this->jobsRepository = $jobsRepository;
+        $this->jobValidator = $jobValidator;
+        $this->companyRepository = $companyRepository;
+    }
+        
+    /**
+     * add job
+     *
+     * @param  mixed $request
+     * @return Response
+     */
+    public function addJob(Request $request): Response
+    {
+        try {
+            $name = $request->get('name');
+            $this->jobValidator->nameIsValid($name);
+            $companyId = (int) $request->get('company_id');
+            $company = $this->companyRepository->find($companyId);
+            
+            if (is_null($company)) {
+                throw new \InvalidArgumentException('Could not find company with id: ' . $companyId);
+            }
+
+            $job = new Jobs();
+            $job->setName($request->get('name'));
+            $job->setCompanyId($company);
+            $job->setDescription($request->get('description'));
+            $job->setCreatedAt(new DateTimeImmutable());
+            $jobSaved = $this->jobsRepository->save($job);
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'Saved new job' => $jobSaved,
+                        'error' => false,
+                    ]
+                ]
+            );
+        } catch (\Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * get all jobs from the database
+     *
+     * @param  mixed $doctrine
+     * @return Response
+     */
+    public function readJobs(): Response
+    {
+
+        $rows = $this->jobsRepository->select();
+        return new JsonResponse([
+            $rows
```

Comment:
the result must be of this format:
`  'results' => [
                    'jobs' => [
                        'id' => $id
                        'name' => $name,
                    'company' => [
                        'id' = $companyId,
                        'name'  => $companyName
                     ]
                    ],
                    'error' => false,
                ]`

We need to get the job company related information too. For this the repository query builder needs a little change. I have added a comment there too.